pkg/berglas/list.go

```diff
@@ -15,15 +15,51 @@
 package berglas
 
 import (
+	"cloud.google.com/go/storage"
 	"context"
+	"fmt"
 	"sort"
+	"time"
 
 	"github.com/pkg/errors"
 	"google.golang.org/api/iterator"
 )
 
+type Secret struct {
+	// Name of the secret
+	Name string
+
+	// UpdatedAt indicates when a secret was last updated
+	UpdatedAt time.Time
+
+	// Generation indicates a secret's version
+	Generation int64
+}
+
+func (s Secret) String() string {
+	return fmt.Sprintf("%s (updated %s) (generation %d)",
+		s.Name, s.UpdatedAt.Local(), s.Generation)
+}
+
+type SecretSlice []Secret
+
+func (s SecretSlice) Len() int {
+	return len(s)
+}
+
+func (s SecretSlice) Less(i, j int) bool {
+	if s[i].Name == s[j].Name {
+		return s[i].Generation > s[j].Generation
+	}
+	return s[i].Name > s[j].Name
+}
+
+func (s SecretSlice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
 // List is a top-level package function for listing secrets.
-func List(ctx context.Context, i *ListRequest) ([]string, error) {
+func List(ctx context.Context, i *ListRequest) (SecretSlice, error) {
```

Comment:
Again, I'm apprehensive to changing the function contract here

Reply:
Yeah I can either change this to `ListResponse` which includes `SecretSlice` as a member field, or just add a new method `ListExtended` (or something like that). Which do you prefer?

Reply:
Let's do a `ListResponse` object to be consistent.

```go
type ListResponse struct {
  Secrets []Secret
}
```

Make `SecretSlice` `secretSlice` and keep it internal only, since it's only used for sorting (you can sort it and then coerce back to `[]Secret`).

We could leave it public if you feel strongly, but I still think the type returned on the struct should be `[]Secret`.

Reply:
Agreed, `[]Secret` is better.